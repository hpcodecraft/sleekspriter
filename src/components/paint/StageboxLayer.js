import React from "react";
import PropTypes from "prop-types";
import { LayerCanvas } from "../canvases";

class StageboxLayer extends React.Component {
  render() {
    const htmlId = `StageBoxLayer-${this.props.layer.id}`,
      display = this.props.layer.visible === true ? "block" : "none",
      style = {
        zIndex: this.props.layer.z,
        opacity: this.props.layer.opacity / 100,
        display,
      };

    return (
      <div id={htmlId} className="Layer" style={style}>
        <LayerCanvas
          ref={n => (this.layerCanvas = n)}
          frame={this.props.layer.frame}
          layer={this.props.layer.id}
          size={this.props.size}
          zoom={this.props.zoom}
          pixels={this.props.pixels}
        />
      </div>
    );
  }
}

StageboxLayer.propTypes = {
  layer: PropTypes.object.isRequired,
  size: PropTypes.object.isRequired,
  zoom: PropTypes.number.isRequired,
  pixels: PropTypes.object,
};

export default StageboxLayer;
