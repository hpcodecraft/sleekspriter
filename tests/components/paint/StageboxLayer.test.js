import StageboxLayer from "components/paint/StageboxLayer";

describe("StageboxLayer", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      layer: {
        id: 1,
        frame: 1,
        visible: true,
        opacity: 100,
        z: 1,
        name: "Layer 1",
      },
      size: {
        height: 32,
        width: 32,
      },
      zoom: 10,
    };
    wrapper = shallow(<StageboxLayer {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
