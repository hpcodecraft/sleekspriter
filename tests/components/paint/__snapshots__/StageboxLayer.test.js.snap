// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageboxLayer should render correctly 1`] = `"<div id=\\"StageBoxLayer-1\\" class=\\"Layer\\" style=\\"z-index:1;opacity:1;display:block;\\"><canvas width=\\"320\\" height=\\"320\\" style=\\"width:320px;height:320px;\\"></canvas></div>"`;
